lib/src/main/java/io/ably/lib/transport/WebSocketTransport.java

```diff
@@ -112,7 +113,12 @@ public void send(ProtocolMessage msg) throws AblyException {
                     Log.v(TAG, "send(): " + new String(ProtocolSerializer.writeJSON(msg)));
                 wsConnection.send(ProtocolSerializer.writeJSON(msg));
             }
-        } catch (Exception e) {
+        }
+        catch (WebsocketNotConnectedException e){
+            AblyException ablyException = AblyException.fromThrowable(e);
+            connectListener.onTransportUnavailable(this, ablyException.errorInfo);
```

Comment:
For what it's worth, the `ablyException` local variable is pretty superfluous here - I would just use `AblyException.fromThrowable(e)` at the call site. 🤷‍♂️ 

Also, would it not be a good idea to check whether `connectListener` is `null` here?

Also, are there any threading considerations (i.e. contract with consumers) made around `ConnectListener` that might suggest calling it synchronously here might cause further issues?